network/p2pMetainfo_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2019-2025 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package network
+
+import (
+	"encoding/binary"
+	"net/http"
+	"testing"
+
+	"github.com/algorand/go-algorand/logging"
+	"github.com/algorand/go-algorand/test/partitiontest"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+// MockStream is a io.ReaderWriter testing mock
+type MockStream struct {
+	mock.Mock
+}
+
+func (m *MockStream) Read(p []byte) (n int, err error) {
+	args := m.Called(p)
+	arg0 := args.Get(0).([]byte)
+	copy(p, arg0)
+	return len(arg0), args.Error(1)
+}
+
+func (m *MockStream) Write(p []byte) (n int, err error) {
+	args := m.Called(p)
+	return args.Int(0), args.Error(1)
+}
+
+func TestReadPeerMetaHeaders(t *testing.T) {
```

Comment:
This doesn't have any testing of bad data on the stream. Is the assumption that bad data isn't very dangerous, it'll just cause a error and the stream to close if it's too short, or if the length bytes are wrong, or whatever?

Reply:
Fair, added test cases covering all the error branches